### PR TITLE
add support for SRV records (python)

### PR DESCRIPTION
**Pull request by Sch8ill**
## Proposed Changes (python)
- add support for SRV records
- bump version to 2.5.0
- add dnspython as dependency in pyproject.toml
- mark this package as statically type (PEP-0561)
- fix small bug from prior pull request

I'm so sorry for the extra effort
I think it would be best if the pull request would be squashed because of the many commits.

### CODE REVIEW

File: Python/minestat/__init__.py [hunk 1]
```diff
@@ -144,23 +147,30 @@ def __str__(self) -> str:
...
     self.address: str = address
     """hostname or IP address of the Minecraft server"""
+
     autoport: bool = False
     if port == 0:
       autoport = True
+
       if query_protocol is SlpProtocols.BEDROCK_RAKNET:
-        self.port = self.DEFAULT_BEDROCK_PORT
+        port = self._resolve_srv_record(self.address, proto="udp")
```
Comment (mindsolve) [inline Python/minestat/__init__.py:164]: There is currently no support for SRV records in Minecraft Bedrock: https://feedback.minecraft.net/hc/en-us/community/posts/360016429012-SRV-Record-Support-Bedrock-Edition-

**Review by mindsolve — CHANGES_REQUESTED** [review]
Hi @Sch8ill,

Thank you very much for the contribution!
Sorry that I have so many comments 😉 - please do not be discouraged to further submit changes!

The two major points:
- There is [no support for SRV records in Minecraft Bedrock](https://feedback.minecraft.net/hc/en-us/community/posts/360016429012-SRV-Record-Support-Bedrock-Edition-), so all logic regarding that diverges from the actual way the client resolves the server address and is not needed.
- You can replace your "is IP address" check with the system module `ipaddress`

After that's changed, looks good!

Best regards,
MindSolve

**Comment by Sch8ill** [comment]
All problems are resolved.
i'm very sorry for this mess, this pull request really needs to be squashed.

**Review by mindsolve — APPROVED** [review]
Looks good :)

**Pull request merged**